omod/src/main/java/org/openmrs/module/fhir2/web/filter/ClientAuthenticationFilter.java

```diff
@@ -42,22 +42,28 @@ public void init(FilterConfig filterConfig) throws ServletException {
 	
 	@Override
 	public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain) throws IOException, ServletException {
-		super.doFilter(req, res, chain);
-		if (!res.isCommitted() && req instanceof HttpServletRequest) {
+		if (req instanceof HttpServletRequest) {
 			HttpServletRequest httpRequest = (HttpServletRequest) req;
-			if (httpRequest.getRequestedSessionId() != null && !httpRequest.isRequestedSessionIdValid()) {
-				Context.logout();
-			}
-			if (httpRequest.getAttribute(KeycloakAccount.class.getName()) != null) {
-				OidcKeycloakAccount account = (OidcKeycloakAccount) httpRequest
-				        .getAttribute(KeycloakAccount.class.getName());
-				Context.becomeUser(account.getPrincipal().getName());
-			} else {
-				HttpServletResponse httpResponse = (HttpServletResponse) res;
-				httpResponse.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Not authenticated");
-				return;
+			if (!httpRequest.getAuthType().contains("BASIC")) {
+				
+				super.doFilter(req, res, chain);
+				if (httpRequest.getRequestedSessionId() != null && !httpRequest.isRequestedSessionIdValid()) {
+					Context.logout();
+				}
+				if (httpRequest.getAttribute(KeycloakAccount.class.getName()) != null) {
+					OidcKeycloakAccount account = (OidcKeycloakAccount) httpRequest
+					        .getAttribute(KeycloakAccount.class.getName());
+					Context.becomeUser(account.getPrincipal().getName());
+				} else {
+					if (!res.isCommitted()) {
+						HttpServletResponse httpResponse = (HttpServletResponse) res;
+						httpResponse.sendError(HttpServletResponse.SC_UNAUTHORIZED, "Not authenticated");
+						return;
+					}
+				}
 			}
 		}
+		
```

Comment:
@ibacher This gives an NPE but don't know why it makes sense to me. The super function will always commit to response that why I thought to write something like this.

Error log
```
SEVERE: Servlet.service() for servlet [openmrs] in context with path [/openmrs] threw exception
java.lang.NullPointerException
	at org.openmrs.module.fhir2.web.filter.ClientAuthenticationFilter.doFilter(ClientAuthenticationFilter.java:47)
	at org.openmrs.module.web.filter.ModuleFilterChain.doFilter(ModuleFilterChain.java:71)
	at org.openmrs.module.referenceapplication.filter.RequireLoginLocationFilter.doFilter(RequireLoginLocationFilter.java:93)
	at org.openmrs.module.web.filter.ModuleFilterChain.doFilter(ModuleFilterChain.java:71)
	at org.openmrs.module.web.filter.ModuleFilter.doFilter(ModuleFilter.java:57)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:241)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:208)

```

Reply:
I'm not sure what part of the code this is referring to exactly, because I think the line numbers have shifted?